common/djangoapps/entitlements/tests/test_entitlements_api.py

```diff
@@ -0,0 +1,204 @@
+import json
+import unittest
+import uuid
+
+from django.conf import settings
+from django.core.urlresolvers import reverse
+from xmodule.modulestore.tests.django_utils import ModuleStoreTestCase
+from xmodule.modulestore.tests.factories import CourseFactory
+
+from entitlements.models import CourseEntitlement
+from openedx.core.lib.token_utils import JwtBuilder
+from student.tests.factories import CourseEnrollmentFactory, UserFactory
+
+
+@unittest.skipUnless(settings.ROOT_URLCONF == 'lms.urls', 'Test only valid in lms')
+class EntitlementsTest(ModuleStoreTestCase):
+    """
+    Entitlements API/View Tests
+    """
+    USERNAME = 'Bob'
+    ENABLED_CACHES = ['default']
+    COURSE_RUN_ID = 'some/great/course'
+
+    def setUp(self):
+        super(EntitlementsTest, self).setUp()
+        self.user = UserFactory(is_staff=True)
+        self.course = CourseFactory.create(org='TestX', course='TS101', run='T1')
+        # self.entitlements_url = '/api/entitlements/v1/entitlements/'
+        self.entitlements_url = reverse('entitlements_api:api:entitlements-list')
+        self.entitlements_uuid_path = 'entitlements_api:api:entitlements-list'
+        self.course_uuid = str(uuid.uuid4())
+
+    def tearDown(self):
+        self.client.logout()
+
+    def _setup_header(self, user):
+        """
+        Get a header containing the Authentication Token
+        """
+        scopes = ['email', 'profile']
+        expires_in = settings.OAUTH_ID_TOKEN_EXPIRATION
+        token = JwtBuilder(user).build_token(scopes, expires_in)
+        return {
+            'HTTP_AUTHORIZATION': 'JWT ' + token
+        }
+
+    def _get_data_set(self, user):
+        """
+        Get a basic data set for an entitlement
+        """
+        return {
+            "user": user.username,
+            "mode": "verified",
+            "course_uuid": self.course_uuid,
+            "order_number": "EDX-1001"
+        }
+
+    def _add_entitlement(self, user, order_number):
+        """
+        Add an entitlement to the database
+        """
+        entitlement = CourseEntitlement()
+        entitlement.course_uuid = uuid.uuid4()
+        entitlement.mode = 'verified'
+        entitlement.user = user
+        entitlement.order_number = order_number
+        entitlement.save()
+        return entitlement
+
+    def test_auth_required(self):
+        headers = {}
+        response = self.client.get(self.entitlements_url, **headers)
+        self.assertEqual(response.status_code, 401)
+
+    def test_staff_user_required(self):
+        headers = {}
+        not_staff_user = UserFactory()
+        self.client.login(username=not_staff_user.username, password=UserFactory._DEFAULT_PASSWORD)
+        response = self.client.get(self.entitlements_url, **headers)
+        self.assertEqual(response.status_code, 403)
```

Comment:
We now use `py.test` for edx-platform. Ideally all tests should be written using `assert` rather than the unittest helper methods. See https://docs.pytest.org/en/latest/assert.html.